### ISSUE TITLE
J.MethodInvocation. getMethodType is null()

### ISSUE THREAD

**Issue by wendigo** [openrewrite/rewrite-testing-frameworks#341]
I'm trying to rewrite a recipe to migrate from TestNG to AssertJ.

My recipe is not matching particular occurrences like this:

```
package io.trino.plugin.jdbc;

import org.testng.annotations.Test;

import static io.trino.spi.block.TestingSession.SESSION;
import static org.testng.Assert.assertNotNull;

public class TestRetryingConnectionFactory
{
    @Test
    public void testSimplyReturnConnection()
            throws Exception
    {
        ConnectionFactory factory = new RetryingConnectionFactory(null);
        assertNotNull(factory.openConnection(SESSION));
    }
}
```

It's working when `assertNotNull(factory.openConnection(SESSION));` is replaced with `assertNotNull(factory.openConnection(null));`

In a `public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {`

method.getMethodType() is null when `factory.openConnection(SESSION)` and 
method.getMethodType() is not null (correctly identifies the type as `org.testng.Assert{name=assertNotNull,return=void,parameters=[java.lang.Object]}`)

`SESSION` const comes from another module (being a `test-jar` dependency if that matters). If I replace that `SESSION` with a const coming from another `test-jar` dependency it works correctly identifying a type.

**Comment by timtebeek** [comment]
Hi @wendigo ; I've move this issue to rewrite-testing-frameworks, as I'd be interested to hear if you'd like to contribute this here as a recipe with a pull request. Fully up to you of course, I'll try to answer your question just the same. :)

The missing types are usually an indication that either your recipe-unit-tests, or recipe itself lack a `.classpath(...)` entry to the `JavaParser.Builder`, or `JavaTemplate.Builder`. We're [looking to improve the docs](https://github.com/openrewrite/rewrite-docs/issues/170), as this [also came up on another issue recently](https://github.com/openrewrite/rewrite/issues/3260#issuecomment-1556136720). The changes I made there are likely similar to what you need here: https://github.com/starksm64/tck-rewrite-tools/pull/1/files

If that does not help, then please share your recipe, and recipe-unit tests such that we can have a look. Ideally as a pull request if that's something you're comfortable with, but a separate Git repository, zip upload, text blocks here or privately shared code could work as well. For the latter it's best to go through [our public Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA).

**Comment by wendigo** [comment]
@timtebeek I don't think this is the case, it works in the same file for other occurrences but for some, it's not.

**Comment by knutwannheden** [comment]
@wendigo That sounds strange. If you could share the code that would be very helpful.

**Comment by timtebeek** [comment]
While we'd very much like to help you, it's not possible without further details. I hope you don't mind me closing this issue until we have those details, to clear out our backlog from unactionable items a bit. Feel free to chime back in with your recipe to help us debug!

**Comment by wendigo** [comment]
I’ve dropped the idea of using this tool in favor of refaster

**Comment by wendigo** [comment]
Refaster is painfully slow but it’s much easier to write simple rules by giving it examples. If you can make it faster then the future is bright for this project :)

**Comment by timtebeek** [comment]
Yes the ease with which you can develop simple recipes is what we liked about it as well; and we hope to build upon the work of others by allowing you to run those through OpenRewrite directly. Running the recipes ought to be quick, but building up the LST model of your code can still take a while. We'll have to see how things pan out in the end.

**Comment by rickie** [comment]
Thanks for tagging @timtebeek :smile:!

@wendigo Let us know if you encounter any issue or when something is missing for the migration when you start using Error Prone Support.

> Refaster is painfully slow but it’s much easier to write simple rules by giving it examples.

We are working on speeding up Refaster in Error Prone Support, but it'll probably take a while before we finish that.
